packages/optimizely-sdk/lib/core/custom_attribute_condition_evaluator/index.ts

```diff
@@ -15,29 +15,32 @@
  ***************************************************************************/
 import { sprintf } from '@optimizely/js-sdk-utils';
 
+import { LoggerFacade } from '@optimizely/js-sdk-logging';
+import { Condition, UserAttributes } from '../../shared_types';
+
 import { isNumber, isSafeInteger } from '../../utils/fns';
 import {
   LOG_LEVEL,
   LOG_MESSAGES,
 } from '../../utils/enums';
 
-var MODULE_NAME = 'CUSTOM_ATTRIBUTE_CONDITION_EVALUATOR';
+const MODULE_NAME = 'CUSTOM_ATTRIBUTE_CONDITION_EVALUATOR';
 
-var EXACT_MATCH_TYPE = 'exact';
-var EXISTS_MATCH_TYPE = 'exists';
-var GREATER_THAN_MATCH_TYPE = 'gt';
-var LESS_THAN_MATCH_TYPE = 'lt';
-var SUBSTRING_MATCH_TYPE = 'substring';
+const EXACT_MATCH_TYPE = 'exact';
+const EXISTS_MATCH_TYPE = 'exists';
+const GREATER_THAN_MATCH_TYPE = 'gt';
+const LESS_THAN_MATCH_TYPE = 'lt';
+const SUBSTRING_MATCH_TYPE = 'substring';
 
-var MATCH_TYPES = [
+const MATCH_TYPES = [
   EXACT_MATCH_TYPE,
   EXISTS_MATCH_TYPE,
   GREATER_THAN_MATCH_TYPE,
   LESS_THAN_MATCH_TYPE,
   SUBSTRING_MATCH_TYPE,
 ];
 
-var EVALUATORS_BY_MATCH_TYPE = {};
+const EVALUATORS_BY_MATCH_TYPE = {};
```

Comment:
Let's add a type annotation on this object - otherwise it seems reading a property of it results in `any` (check the type of `evaluatorForMatch` on line 76). How about:
```suggestion
type ConditionEvaluator = (condition: Condition, userAttributes: UserAttributes) => boolean | null;
const EVALUATORS_BY_MATCH_TYPE: { [conditionType: string]: ConditionEvaluator | undefined } = {};
```